src/txn/txn.c

```diff
@@ -664,7 +703,11 @@ __txn_commit_timestamp_validate(WT_SESSION_IMPL *session)
 				op_timestamp = txn->commit_timestamp;
 			if (op_timestamp < upd->start_ts)
```

Comment:
This code is now assuming that `WT_TXN_TS_DURABLE_KEYS` implies `WT_TXN_TS_COMMIT_KEYS`. Either the commit-related flag should be set when we parse the durable-related one in `bt_handle.c` (and a comment/documentation added stating that) or if they're separate, this `if` needs to also be conditional on `WT_TXN_TS_COMMIT_KEYS` being set.